src/serac/infrastructure/profiling.hpp

```diff
@@ -71,4 +85,37 @@ void initializeCaliper(const std::string& options = "");
  */
 void terminateCaliper();
 
+/**
+ * @brief Caliper metadata methods cali_set_global_(double|int|string|uint)_byname()
+ */
+template <typename T>
+void setCaliperMetaData(const std::string& name, T data);
+
+/**
+ * @brief Adds double with given name to caliper metadata
+ */
+template <>
+void setCaliperMetaData<double>(const std::string& name, double data);
+
+/**
+ * @brief Adds int with given name to caliper metadata
+ */
+
+template <>
+void setCaliperMetaData<int>(const std::string& name, int data);
+
+/**
+ * @brief Adds string with given name to caliper metadata
+ */
+
+template <>
+void setCaliperMetaData<const char*>(const std::string& name, const char* data);
+
+/**
+ * @brief Adds unsigned int with given name to caliper metadata
+ */
+
+template <>
+void setCaliperMetaData<unsigned int>(const std::string& name, unsigned int data);
+
```

Comment:
I'm not sure we need these forward declarations for each specific type, unless it's just for doxygen.

It seems like setCaliperMetaData could be an overload set, e.g.
```cpp
void setCaliperMetaData(const std::string& name, double data);
void setCaliperMetaData(const std::string& name, int data);
void setCaliperMetaData(const std::string& name, unsigned int data);
...
```
or a function template (without forward declaring the explicit specializations)
```cpp
template <typename T>
void setCaliperMetaData(const std::string& name, T data);
```

I also think metadata is a single word, so the capital D looks out of place to me.